src/xmpp/avatar.c

```diff
@@ -339,10 +339,21 @@ _avatar_request_item_result_handler(xmpp_stanza_t* const stanza, void* const use
 
     // if we shall open it
     if (g_hash_table_contains(shall_open, from_attr)) {
-        gchar* argv[] = { prefs_get_string(PREF_AVATAR_CMD), filename->str, NULL };
-        if (!call_external(argv, NULL, NULL)) {
-            cons_show_error("Unable to display avatar: check the logs for more information.");
+        gchar* cmdtemplate = prefs_get_string(PREF_AVATAR_CMD);
+
+        if (cmdtemplate == NULL) {
+            cons_show_error("No default `avatar open` command found in executables preferences.");
+        }
+        else {
+            gchar **argv = format_call_external_argv(cmdtemplate, NULL, filename->str);
+
+            if (!call_external(argv, NULL, NULL)) {
+                cons_show_error("Unable to display avatar: check the logs for more information.");
+            }
+
+            g_strfreev(argv);
```

Comment:
```suggestion
        gchar* argv[] = { prefs_get_string(PREF_AVATAR_CMD), filename->str, NULL };
        if (argv[0] == NULL)
            cons_show_error("No default `avatar open` command found in executables preferences.");
        else if (!call_external(argv, NULL, NULL)) {
            cons_show_error("Unable to display avatar: check the logs for more information.");
```

I like the extended check but TBH I'd prefer to keep the old style and simply do the check as suggested

Reply:
I mean this one as will render it impossible to specify "gio open %p" because making it first argument in an array makes `gio open` an entire one argument. It doesn't push `open` as second one. `format_call_external_argv` on the other hand handles spaces.

I think that was the reason why the issue owner had the error initially which then coincidentally resulted in a crash due to another reason (undefined behavior). `gio open` was just not found as a program.

Reply:
Though the failed code style check made me cry a little bit 😢

Reply:
> I mean this one as will render it impossible to specify "gio open %p" because making it first argument in an array makes gio open an entire one argument. It doesn't push open as second one. format_call_external_argv on the other hand handles spaces.

Oh alright, that makes a lot of sense!